stdlib/public/core/RangeReplaceableCollection.swift

```diff
@@ -459,8 +459,6 @@ extension RangeReplaceableCollection {
     }
       
     if done == nil {
-      let approximateCapacity = self.count + newElements.underestimatedCount
-      self.reserveCapacity(approximateCapacity)
```

Comment:
It may be a good idea to add an explicit comment about this gotcha, to help prevent this from happening again.

```suggestion
    // Note: This used to call `reserveCapacity` here, but that can dramatically
    // slow down clients that repeatedly call `append(contentsOf:)`. Do not
    // re-add that. (The usual implementation of `reserveCapacity` for
    // contiguous collections disables exponential growth for the backing store,
    // ruining the amortized O(*m*) complexity that we promise above.)

```